### ISSUE TITLE
Execution gets idle for few seconds in between user injection and writing request

### ISSUE THREAD

**Issue by priyanshus** [gatling/gatling#3912]
I am running a test on my. Mac 10.14.6 via Maven Plugin. I have noticed that the execution gets idle for few seconds before writing the request `i.g.h.c.i.Http2AppHandler`. The log says that user has been injected at `08:54:47.706 [DEBUG]` and actual request hits to server at `08:54:54.167`, so there is a delay of 7 second in between user injection and hitting the server and that spoils the report. Report shows response time 7XXX where XXX was actual time taken by server but the additional 7s is because of gatling execution delay.

**Execution Log**
Simulation com.tw.performance.CheckShippingDetailsSimulation started...
08:54:47.593 [INFO ] i.g.c.s.w.ConsoleDataWriter - Initializing
08:54:47.594 [INFO ] i.g.c.s.w.LogFileDataWriter - Initializing
08:54:47.599 [INFO ] i.g.c.s.w.ConsoleDataWriter - Initialized
08:54:47.621 [INFO ] i.g.c.s.w.LogFileDataWriter - Initialized
**08:54:47.640 [DEBUG] i.g.c.c.i.o.OpenWorkload - Injecting 1 users in scenario Check Shipping Details, continue=false**
08:54:47.641 [INFO ] i.g.c.c.i.Injector - StoppedInjecting
08:54:47.687 [DEBUG] i.g.h.e.t.HttpTxExecutor - Sending request=Post Login uri=https://gatling.io/: scenario=Check Shipping Details, userId=1
08:54:47.706 [DEBUG] i.g.c.c.i.o.OpenWorkload - Start user #1
**08:54:52.756 [DEBUG] i.n.c.DefaultChannelId - -Dio.netty.processId: 6903 (auto-detected)**
08:54:52.758 [DEBUG] i.n.u.NetUtil - -Djava.net.preferIPv4Stack: true
08:54:52.758 [DEBUG] i.n.u.NetUtil - -Djava.net.preferIPv6Addresses: false
08:54:52.759 [DEBUG] i.n.u.NetUtil - Loopback interface: lo0 (lo0, 127.0.0.1)
08:54:52.760 [DEBUG] i.n.u.NetUtil - Failed to get SOMAXCONN from sysctl and file /proc/sys/net/core/somaxconn. Default: 128
08:54:52.762 [DEBUG] i.n.c.DefaultChannelId - -Dio.netty.machineId: 88:e9:fe:ff:fe:7e:51:f3 (auto-detected)
08:54:53.286 [DEBUG] i.g.h.c.i.DefaultHttpClient - Installing SslHandler for https://gatling.io/
08:54:53.290 [DEBUG] i.n.u.ResourceLeakDetectorFactory - Loaded default ResourceLeakDetector: io.netty.util.ResourceLeakDetector@26cb2fed
08:54:53.725 [DEBUG] i.n.h.s.u.InsecureTrustManagerFactory - Accepting a server certificate: CN=gatling.io
08:54:54.109 [DEBUG] i.g.h.c.i.DefaultHttpClient - Installing Http2Handler for https://gatling.io/
08:54:54.119 [DEBUG] i.n.h.s.SslHandler - [id: 0xeb0b2c7c, L:/192.168.43.36:53850 - R:gatling.io/34.77.87.202:443] HANDSHAKEN: TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
08:54:54.120 [DEBUG] i.g.h.c.i.DefaultHttpClient - ALPN led to HTTP/2 with remote gatling.io
08:54:54.162 [DEBUG] i.g.h.c.p.CoalescingChannelPool - Adding entry IpAndPort{ip=[34, 77, 87, -54], port=443} with subjectAlternativeNames [gatling.io] to coalescing pool
**08:54:54.167 [DEBUG] i.g.h.c.i.Http2AppHandler - Write request WritableRequest{request=DefaultFullHttpRequest(decodeResult: success, version: HTTP/1.1, content: UnpooledByteBufAllocator$InstrumentedUnpooledUnsafeHeapByteBuf(ridx: 0, widx: 0, cap: 0))**

**Report**

================================================================================
---- Global Information --------------------------------------------------------
> request count                                          1 (OK=1      KO=0     )
> min response time                                   7157 (OK=7157   KO=-     )
> max response time                                   7157 (OK=7157   KO=-     )
> mean response time                                  7157 (OK=7157   KO=-     )
> std deviation                                          0 (OK=0      KO=-     )
> response time 50th percentile                       7157 (OK=7157   KO=-     )
> response time 75th percentile                       7157 (OK=7157   KO=-     )
> response time 95th percentile                       7157 (OK=7157   KO=-     )
> response time 99th percentile                       7157 (OK=7157   KO=-     )
> mean requests/sec                                  0.125 (OK=0.125  KO=-     )
---- Response Time Distribution ------------------------------------------------
> t < 800 ms                                             0 (  0%)
> 800 ms < t < 1200 ms                                   0 (  0%)
> t > 1200 ms                                            1 (100%)
> failed                                                 0 (  0%)
================================================================================

I am guessing that the netty process is taking some time as I have performed same test on different urls and all of them shows extra few seconds in report because of this.

**Comment by slandelle** [comment]
I’m almost sure that:
* you’ve disabled Gatling warmUp, which causes this delay to happen in the first user instead of before in the warmUp
* your network configuration is broken and your missing the expected "127.0.0.1       localhost" entry in /etc/hosts. This causes this delay which is actually a reverse DNS lookup timeout on localhost.

**Comment by priyanshus** [comment]
@slandelle Thanks for prompt reply. Yes, I did not have warmUP url in the config. That I changed, but I do have `127.0.0.1   localhost` entry in my hosts. 

However, I figured it out that in mac there is an issue related to netty and it got fixed by adding an entry as `127.0.0.1   localhost <my-mac-name>`.
https://stackoverflow.com/questions/39636792/jvm-takes-a-long-time-to-resolve-ip-address-for-localhost/39698914#39698914

**Comment by slandelle** [comment]
Exactly what I said. But it's not a Netty issue, it's an issue in your Mac set up.

By default, `/etc/hosts` on OSX starts with:

```
##
# Host Database
#
# localhost is used to configure the loopback interface
# when the system is booting.  Do not change this entry.
##
127.0.0.1       localhost
```

For some reason, your file was edited and this entry removed, breaking the system.
